libnetwork/networkdb/networkdb.go

```diff
@@ -585,21 +588,14 @@ func (nDB *NetworkDB) deleteNodeTableEntries(node string) {
 // value. The walk stops if the passed function returns a true.
 func (nDB *NetworkDB) WalkTable(tname string, fn func(string, string, []byte, bool) bool) error {
 	nDB.RLock()
-	values := make(map[string]*entry)
-	nDB.indexes[byTable].Root().WalkPrefix([]byte("/"+tname), func(path []byte, v *entry) bool {
-		values[string(path)] = v
-		return false
-	})
+	root := nDB.indexes[byTable].Root()
 	nDB.RUnlock()
-
-	for k, v := range values {
-		params := strings.Split(k[1:], "/")
+	root.WalkPrefix([]byte("/"+tname), func(path []byte, v *entry) bool {
+		params := strings.Split(string(path[1:]), "/")
 		nid := params[1]
 		key := params[2]
```

Comment:
Not for this PR, but would this be something we could replace for `strings.Cut`, or do we expect more than one `/` (and only need the first two components here)?